src/GShark/Geometry/Mesh.cs

```diff
@@ -548,6 +548,144 @@ private FaceData CountFaceEdges()
         }
 
 
+        //Function that finds point in a cloud of points to a target point
+        public static Point3 CloudClosestPoint(Point3 target, List<Point3> cloud)
+        {
+            var DistancesToTarget_List = new List<double>();
+            foreach (Point3 point in cloud)
+            {
+                double DistanceToTarget = point.DistanceTo(target);
+                DistancesToTarget_List.Add(DistanceToTarget);
+            }
+            double MinDistToTarget = DistancesToTarget_List.Min();
+            Point3 ClosestPoint = cloud[DistancesToTarget_List.IndexOf(MinDistToTarget)];
+            return ClosestPoint;
+        }
+
+        //Function that checks if point is in or out a triangle (boolean result, In - True , Out - False)
+        private bool PointInTriangle(Point3 projection, List<Point3> trianglePoints)
```

Comment:
```suggestion
        private bool IsPointInTriangle(Point3 projection, List<Point3> trianglePoints)
```

Usually a method that answer to a question and return a bool start with `Is`